### DISCUSSION TITLE
PutURL with DiscoveredURLItem & completed URL in URLFrontier

### DISCUSSION THREAD

**Discussion by klockla** [crawler-commons/url-frontier#94] [Q&A] [answered]
Hello,

Currently if we have a completed URL (KnownURL and refetch date = 0) in the frontier, and we send a putURL with a DiscoveredURLItem for the same URL, the implementation (in RockDBService at least, didn't check the others) will reschedule it.

Is this the intended behavior or a bug ?

**Comment by jnioche** [comment]
Hi @klockla, sorry about the late reply, I have just returned from holiday and as you have certainly noticed, I am the only dev on this project. 
The project is not very active and it has been a while since I last looked at it. As far as I can remember, you are correct - a discovered URL should not overwrite the refetch date of a known URL.
Would you be able to write a unit test to illustrate the problem?
PS: I will have a look at the 2 PRs you submitted, thanks for your patience

**Comment by klockla** [comment, marked as answer]
I cannot reproduce the issue anymore but I have a test case for it in PR #101
